Doc/library/re.rst

```diff
@@ -1562,21 +1562,24 @@ search() vs. match()
 
 .. sectionauthor:: Fred L. Drake, Jr. <fdrake@acm.org>
 
-Python offers two different primitive operations based on regular expressions:
-:func:`re.match` checks for a match only at the beginning of the string, while
-:func:`re.search` checks for a match anywhere in the string (this is what Perl
-does by default).
+Python offers different primitive operations based on regular expressions:
+:func:`re.match` checks for a match only at the beginning of the string,
+:func:`re.fullmatch` checks for entire string to be a match. :func:`re.search`
+checks for a match anywhere in the string (this is what Perl does by default).
 
 For example::
 
    >>> re.match("c", "abcdef")    # No match
    >>> re.search("c", "abcdef")   # Match
    <re.Match object; span=(2, 3), match='c'>
+   >>> re.fullmatch("python", "python")  # Match
+   <re.Match object; span=(0, 6), match='python'>
```

Comment:
None of these examples differentiate between `re.fullmatch` and `re.match`

Reply:
Thanks I'll do it

Reply:
what example can we take then to differentiate between them

Reply:
I think you'd need to show an example where `search()` or `match()` match, but `fullmatch()` does not.

Reply:
ok, i'll do that

Reply:
done with the changes, sir